teacherchannel/database.py

```diff
@@ -0,0 +1,160 @@
+        "t_channel_id",
+        BigInteger,
+        ForeignKey(
+            "school_teacherchannel_teacherchannel.t_channel_id", ondelete="CASCADE"
+        ),
+    ),
+    Column(
+        "teacher_id",
+        BigInteger,
+        ForeignKey("school_teacherchannel_teacher.teacher_id", ondelete="CASCADE"),
+    ),
+)
+
+
+class TeacherChannel(database.base):
+    """Subject channels containing teacher channel, synced from the original channel.
+    t_channel is the new teacher channel, o_channel is original."""
+
+    __tablename__ = "school_teacherchannel_teacherchannel"
+
+    t_channel_id = Column(BigInteger, primary_key=True, autoincrement=False)
+    guild_id = Column(BigInteger)
+    o_channel_id = Column(BigInteger, unique=True)
+    teachers = relationship(
+        "Teachers",
+        secondary=association_table,
+        back_populates="channels",
+        cascade="all, delete",
+    )
+
+    @staticmethod
+    def add(
+        guild_id: int, t_channel_id: int, o_channel_id: int, teacher_id: int
+    ) -> TeacherChannel:
+        teacherchannel = (
+            session.query(TeacherChannel)
+            .filter_by(
+                guild_id=guild_id, t_channel_id=t_channel_id, o_channel_id=o_channel_id
+            )
+            .one_or_none()
+        )
+        teacher = Teachers.get(teacher_id)
+        if not teacher:
+            teacher = Teachers(teacher_id=teacher_id)
+        if not teacherchannel:
+            teacherchannel = TeacherChannel(
+                guild_id=guild_id, t_channel_id=t_channel_id, o_channel_id=o_channel_id
+            )
+            teacherchannel.teachers.append(teacher)
+            session.add(teacherchannel)
+        elif teacher_id not in [i.teacher_id for i in teacherchannel.teachers]:
+            teacherchannel.teachers.append(teacher)
+            session.merge(teacherchannel)
+        session.commit()
+        return teacherchannel
+
+    @staticmethod
+    def get_all(guild_id: int) -> List[TeacherChannel]:
+        return session.query(TeacherChannel).filter_by(guild_id=guild_id).all()
+
+    @staticmethod
+    def get(guild_id: int, o_channel_id: int) -> Optional[TeacherChannel]:
+        return (
+            session.query(TeacherChannel)
+            .filter_by(guild_id=guild_id, o_channel_id=o_channel_id)
+            .one_or_none()
+        )
+
+    @staticmethod
+    def get_by_t_channel(guild_id: int, t_channel_id: int) -> Optional[TeacherChannel]:
+        return (
+            session.query(TeacherChannel)
+            .filter_by(guild_id=guild_id, t_channel_id=t_channel_id)
+            .one_or_none()
+        )
+
+    @staticmethod
+    def remove(guild_id: int, t_channel_id: int, teacher_id: int):
+        teacherchannel = (
+            session.query(TeacherChannel)
+            .filter_by(guild_id=guild_id, t_channel_id=t_channel_id)
+            .one_or_none()
+        )
+        # searching for related objects
+        for teacher in teacherchannel.teachers:
```

Comment:
Tohle celé by se mělo řešit přes SQLAlchemy dotazem na konkrétní tabulku s použitím funkce delete()

https://stackoverflow.com/questions/9882358/how-to-delete-rows-from-a-table-using-an-sqlalchemy-query-without-orm